### PR TITLE
Allow use of DiscreteUniform distribution for CategoricalGibbsMetropolis step method

### PR DESCRIPTION
**Pull request by helmutsimon**
The CategoricalGibbsMetropolis step method is the default for the Categorical distribution. The Metropolis step method may not work for Categorical in some circumstances, e,.g. as part of compound distribution. However, CategoricalGibbsMetropolis currently can only be used with Categorical or Bernoulli distributions.

For some models it is necessary to use a DiscreteUniform distribution rather than Categorical, as the k parameter can be very large. This change allows the CategoricalGibbsMetropolis step method to be used for DiscreteUniform distributions, provided that the parameter `lower' is zero.

### CODE REVIEW

File: pymc3/step_methods/metropolis.py [hunk 1]
```diff
@@ -400,9 +400,14 @@ def __init__(self, vars, proposal='uniform', order='random', model=None):
...
                 k = 2
+            elif isinstance(distr, pm.DiscreteUniform):
+                k = draw_values([distr.upper])[0]
+                if np.all(draw_values([distr.lower])):
+                    raise ValueError('Parameter lower must be 0 to use DiscreteUniform' +
+                                     'with CategoricalGibbsMetropolis.')
             else:
                 raise ValueError('All variables must be categorical or binary' +
-                                 'for CategoricalGibbsMetropolis')
+                                 'or discreteuniform for CategoricalGibbsMetropolis')
```
Comment (twiecki) [inline pymc3/step_methods/metropolis.py]:
```suggestion
                                 'or DiscreteUniform for CategoricalGibbsMetropolis')
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### New features
 
+- Allow DiscreteUniform distribution to be used with CategoricalGibbsMetropolis step method.
```
Comment (twiecki) [inline RELEASE-NOTES.md]: Can you move this to the bottom of new features?

File: pymc3/examples/discreteuniform_test.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import numpy as np
+from pymc3 import *
+
+
+def run_MCMC(lower, upper, obs, draws):
```
Comment (twiecki) [inline pymc3/examples/discreteuniform_test.py:5]: We're using pytest for testing, you can check out some other tests but at the minimum this funciton name needs to start with `test_`.

File: pymc3/examples/discreteuniform_test.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import numpy as np
+from pymc3 import *
```
Comment (twiecki) [inline pymc3/examples/discreteuniform_test.py:2]: Can't * import.

File: pymc3/examples/test_DiscreteUniform.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+from numpy import array
```
Comment (twiecki) [inline pymc3/examples/test_DiscreteUniform.py:1]: This should also be in the `tests` directory and probably just added to an existing test file there, like a test_step_methods or something.

Comment (helmutsimon) [reply pymc3/examples/test_DiscreteUniform.py:1]: Makes sense. I have now done this.

File: docs/source/notebooks/SMC2_gaussians.ipynb [hunk 6]
```diff
@@ -53,7 +53,7 @@
     "\n",
     "The previous paragraph is summarized in the next figure, the first subplot show 5 samples (orange dots) at some particular stage. The second subplots show how this samples are reweighted according to the their posterior density (blue Gaussian curve). The third subplot shows the result of running a certain number of Metropolis steps, starting from the _selected/reweighting_ samples in the second subplots, notice how the two samples with the lower posterior density (smaller circles) are discarded and not used to seed Markov chains.\n",
     "\n",
-    "<img src=\"smc.png\" alt=\"SMC stages\"> \n",
+    "<img src=\"https://github.com/pymc-devs/pymc3/raw/master/docs/source/notebooks/smc.png\" alt=\"SMC stages\"> \n",
```
Comment (twiecki) [inline docs/source/notebooks/SMC2_gaussians.ipynb]: this shouldn't show up here.

**Comment by twiecki** [comment]
Needs a rebase.

File: pymc3/examples/test_DiscreteUniform.py (hunk 5 above)
Comment (twiecki) [reply pymc3/examples/test_DiscreteUniform.py:1]: I still these files committed here.

File: pymc3/examples/discreteuniform_test.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import numpy as np
```
Comment (twiecki) [inline pymc3/examples/discreteuniform_test.py:1]: Please remove this file.

**Pull request closed without merging**